flink-runtime/src/test/java/org/apache/flink/runtime/clusterframework/types/LocationPreferenceSlotSelectionStrategyTest.java

```diff
@@ -76,24 +84,27 @@ public void matchNoRequirements() {
 		SlotProfile slotProfile = SlotProfile.noRequirements();
 		Optional<SlotSelectionStrategy.SlotInfoAndLocality> match = runMatching(slotProfile);
 
-		Assert.assertTrue(
-				candidates.stream()
-						.map(SlotSelectionStrategy.SlotInfoAndResources::getSlotInfo)
-						.collect(Collectors.toList())
-						.contains(match.get().getSlotInfo()));
+		Assert.assertTrue(match.isPresent());
+		final SlotSelectionStrategy.SlotInfoAndLocality slotInfoAndLocality = match.get();
+		assertThat(candidates, hasItem(withSlotInfo(slotInfoAndLocality.getSlotInfo())));
+		assertThat(slotInfoAndLocality.getLocality(), is(Locality.UNCONSTRAINED));
 	}
 
 	@Test
-	public void matchPreferredLocationNotAvailable() {
+	public void returnsHostLocalMatchingIfExactTMLocationCannotBeFulfilled() {
 
 		SlotProfile slotProfile = SlotProfile.preferredLocality(resourceProfile, Collections.singletonList(tmlX));
 		Optional<SlotSelectionStrategy.SlotInfoAndLocality> match = runMatching(slotProfile);
 
-		Assert.assertTrue(
-				candidates.stream()
-						.map(SlotSelectionStrategy.SlotInfoAndResources::getSlotInfo)
-						.collect(Collectors.toList())
-						.contains(match.get().getSlotInfo()));
+		Assert.assertTrue(match.isPresent());
+		final SlotSelectionStrategy.SlotInfoAndLocality slotInfoAndLocality = match.get();
+		final SlotInfo slotInfo = slotInfoAndLocality.getSlotInfo();
+		assertThat(candidates, hasItem(withSlotInfo(slotInfo)));
+		assertThat(slotInfoAndLocality.getLocality(), is(Locality.HOST_LOCAL));
```

Comment:
Looks this commit also refines the existing tests to verify slot locality. I think it's better to note it in the commit message.

Reply:
Makes sense. I'll update the commit message.

Reply:
I actually pulled this change out into a separate commit.